### PR TITLE
Retry if sending invite fails

### PR DESCRIPTION
**Pull request by aanil**
No easy way to find out if sending succeeded, so made a function to retry sending twice if an exception occurs during sending an invite.

Can close this if we feel this is overkill.

### CODE REVIEW

File: dds_web/api/user.py [hunk 1]
```diff
@@ -218,11 +218,34 @@ def invite_user(args):
             unit_email=unit_email,
         )
 
-        mail.send(msg)
+        AddUser.send_email_with_retry(msg)
```
Comment (inaod568) [inline dds_web/api/user.py:221]: Not `self`?

File: dds_web/api/user.py [hunk 2]
```diff
@@ -218,11 +218,34 @@ def invite_user(args):
...
         return {"email": new_invite.email, "message": "Invite successful!", "status": 200}
 
+    @staticmethod
+    def send_email_with_retry(msg, times_retried=0):
+        """Send email with retry on exception"""
+
+        try:
+            mail.send(msg)
+        except Exception as err:
+            print("here")
```
Comment (inaod568) [inline dds_web/api/user.py:241]: Remove

**Review by inaod568 — CHANGES_REQUESTED** [review]
I like it! Don't see any reason not to keep this. 

Just remove the `print` and I'll approve. The other comment is not a dealbreaker but (unless I'm missing something and it doesn't work with self for some reason) I think `self` is better.

File: dds_web/api/user.py (hunk 2 above)
Comment (aanil) [reply dds_web/api/user.py]: Oops! Removed.

File: dds_web/api/user.py (hunk 1 above)
Comment (aanil) [reply dds_web/api/user.py:221]: Its a static function, so can't be called with `self`.

Comment (inaod568) [reply dds_web/api/user.py:221]: Riiight nvm

File: dds_web/api/user.py [hunk 3]
```diff
@@ -218,11 +218,33 @@ def invite_user(args):
...
         # TODO: Format response with marshal with?
         return {"email": new_invite.email, "message": "Invite successful!", "status": 200}
 
+    @staticmethod
+    def send_email_with_retry(msg, times_retried=0):
+        """Send email with retry on exception"""
+
+        try:
+            mail.send(msg)
+        except Exception as err:
```
Comment (inaod568) [inline dds_web/api/user.py:240]: Change this to the `SMTPException` and we're all good 👍🏻

Comment (aanil) [reply dds_web/api/user.py]: Done!

**Review by inaod568 — APPROVED** [review]
Looks good!

**Pull request merged**